File: hw/paillacho/U3/HW20Composite/Composite/src/main/java/ec/edu/espe/composite/model/President.java
package ec.edu.espe.composite.model;

/**
 *
 * @author Carlos Paillacho
 */
public class President extends Supervisor {
    private static final President president = new President();

    /**
     * Private constructor sets the title.
     */
    private President() {
        title = "President";
    }

    /**
     * Retrieves the single President instance and sets its name.
     *
     * @param aName the president's name
     * @return the singleton President instance
     */
    public static President getPresident(String aName) {
        president.name = aName;
        return president;
    }

    @Override
    public void stateName() {
        // Additional behavior for President if needed
        super.stateName();
    }
}

File: hw/chicaiza/u3/HW20Composite/CompositePatternJS/ec/edu/espe/model/President.js
import { Supervisor } from './Supervisor.js';

export class President extends Supervisor {
  static instance = null;

  constructor(name = "") {
    if (President.instance) return President.instance;
    super(name, "President");
    President.instance = this;
  }

  static getPresident(name) {
    if (!President.instance) {
      President.instance = new President(name);
    } else {
      President.instance.name = name;
    }
    return President.instance;
  }

  stateName() {
    super.stateName();
  }
}
